packages/monitor-v2/src/monitor-og/oSnapAutomation.ts

```diff
@@ -297,51 +312,42 @@ export const filterUnblockedProposals = async (
   });
 };
 
-// Verifies proposals before they are proposed on-chain.
+// Verifies proposals before they are proposed on-chain. Caller must ensure that proposalsOriginal has all proposal id
+// keys from proposalsExpanded.
 export const filterVerifiedProposals = async (
-  proposals: SnapshotProposalExpanded[],
+  proposalsExpanded: SnapshotProposalExpanded[],
+  proposalsOriginal: Record<string, SnapshotProposalGraphql>,
   supportedModules: SupportedModules,
   params: MonitoringParams
 ): Promise<SnapshotProposalExpanded[]> => {
-  // Convert expanded proposals back to GraphqlData format as this is used in SnapshotVerification.
-  const graphqlData: GraphqlData = {
-    proposals: proposals.map((proposal) => {
-      const { safe, ...clonedObject } = proposal;
-      return { ...clonedObject, plugins: { safeSnap: { safes: [safe] } } };
-    }),
-  };
-
   // Verify all potential proposals.
   const lastTimestamp = await getBlockTimestamp(params.provider, params.blockRange.end);
   const verifiedProposals = (
     await Promise.all(
-      graphqlData.proposals.map(async (proposal) => {
+      proposalsExpanded.map(async (proposal) => {
+        const originalProposal = proposalsOriginal[proposal.id];
+        assert(originalProposal !== undefined, `Original snapshot proposal not found for id: ${proposal.id}`);
+
         // Check that the proposal was approved properly on Snapshot assuming we are at the end block timestamp.
-        const voteOutcomVerified = verifyVoteOutcome(proposal, lastTimestamp, 0).verified;
+        const voteOutcomeVerified = verifyVoteOutcome(originalProposal, lastTimestamp, 0).verified;
 
         // Check that proposal is hosted on IPFS and its content matches.
-        const ipfsVerified = (await verifyIpfs(proposal, params)).verified;
+        const ipfsVerified = (await verifyIpfs(originalProposal, params)).verified;
```

Comment:
This was failing for proposals with multiple safes as we had split proposals by safes, but IPFS contained all safes.